File: bee/beginners/1021.py
'''
1021 - NOTAS E MOEDAS

Leia um valor de ponto flutuante com duas casas decimais. Este valor representa um valor monetário. A seguir, calcule o menor número de notas e moedas possíveis no qual o valor pode ser decomposto. As notas consideradas são de 100, 50, 20, 10, 5, 2. As moedas possíveis são de 1, 0.50, 0.25, 0.10, 0.05 e 0.01. A seguir mostre a relação de notas necessárias.

Entrada
O arquivo de entrada contém um valor de ponto flutuante N (0 ≤ N ≤ 1000000.00).

Saída
Imprima a quantidade mínima de notas e moedas necessárias para trocar o valor inicial, conforme exemplo fornecido.

Obs: Utilize ponto (.) para separar a parte decimal.
'''
valor = float(input())

print("NOTAS:")
notas = valor // 100
print(f"{notas:.0f} nota(s) de R$ 100.00")
valor = valor % 100

notas = valor // 50
print(f"{notas:.0f} nota(s) de R$ 50.00")
valor = valor % 50

notas = valor // 20
print(f"{notas:.0f} nota(s) de R$ 20.00")
valor = valor % 20

notas = valor // 10
print(f"{notas:.0f} nota(s) de R$ 10.00")
valor = valor % 10

notas = valor // 5
print(f"{notas:.0f} nota(s) de R$ 5.00")
valor = valor % 5

notas = valor // 2
print(f"{notas:.0f} nota(s) de R$ 2.00")
valor = valor % 2

print("MOEDAS:")
notas = valor // 1
print(f"{notas:.0f} moeda(s) de R$ 1.00")
valor = valor % 1

notas = valor // 0.5
print(f"{notas:.0f} moeda(s) de R$ 0.50")
valor = valor % 0.5

notas = valor // 0.25
print(f"{notas:.0f} moeda(s) de R$ 0.25")
valor = valor % 0.25

notas = valor // 0.10
print(f"{notas:.0f} moeda(s) de R$ 0.10")
valor = valor % 0.10

notas = valor // 0.05
print(f"{notas:.0f} moeda(s) de R$ 0.05")
valor = valor % 0.05

notas = valor / 0.01
print(f"{notas:.0f} moeda(s) de R$ 0.01")
valor = valor % 0.01

File: beecrowd/cpp/beginners/1021.cpp
#include <iostream>
#include <cmath>

int main()
{
    double n, valor, resto;
    std::cin >> n;

    std::cout << "NOTAS:" << std::endl;
    valor = n / 100;
    resto = std::fmod(n, 100);
    std::cout << (int)valor << " nota(s) de R$ 100.00" << std::endl;

    valor = resto / 50;
    resto = std::fmod(resto, 50);
    std::cout << (int)valor << " nota(s) de R$ 50.00" << std::endl;

    valor = resto / 20;
    resto = std::fmod(resto, 20);
    std::cout << (int)valor << " nota(s) de R$ 20.00" << std::endl;

    valor = resto / 10;
    resto = std::fmod(resto, 10);
    std::cout << (int)valor << " nota(s) de R$ 10.00" << std::endl;

    valor = resto / 5;
    resto = std::fmod(resto, 5);
    std::cout << (int)valor << " nota(s) de R$ 5.00" << std::endl;

    valor = resto / 2;
    resto = std::fmod(resto, 2);
    std::cout << (int)valor << " nota(s) de R$ 2.00" << std::endl;

    std::cout << "MOEDAS:" << std::endl;
    valor = resto / 1;
    resto = std::fmod(resto, 1);
    std::cout << (int)valor << " moeda(s) de R$ 1.00" << std::endl;

    valor = resto / 0.5;
    resto = std::fmod(resto, 0.5);
    std::cout << (int)valor << " moeda(s) de R$ 0.50" << std::endl;

    valor = resto / 0.25;
    resto = std::fmod(resto, 0.25);
    std::cout << (int)valor << " moeda(s) de R$ 0.25" << std::endl;

    valor = resto / 0.10;
    resto = std::fmod(resto, 0.10);
    std::cout << (int)valor << " moeda(s) de R$ 0.10" << std::endl;

    valor = resto / 0.05;
    resto = std::fmod(resto, 0.05);
    std::cout << (int)valor << " moeda(s) de R$ 0.05" << std::endl;

    valor = std::round(resto / 0.01);
    std::cout << (int)valor << " moeda(s) de R$ 0.01" << std::endl;
}
